Remove auth token and port openstack module_utils changes to v2 tree

emonty opened the pull request.

emonty committed:
Remove auth_token parameter

It turns out that this can actually already be handled by the existing
auth plugin framework and does not need its own parameter. Remove before
it sees usage and causes confusion.

emonty committed:
Port openstack module_utils changes to v2 branch

emonty committed:
Remove state from central argument list

There is an old PR that shows a great use case for having a different
set of states for the server module. Before the other modules start
being in real use, pull this out so that we don't get ourselves into a
pickle.

emonty commented:
@bcoca we should pop this one in around the same time as the ansible-modules-extras one. I just read through the outstanding PRs on the old modules, and there is one feature someone wanted that's really good that will need this and the corresponding change there (which is already pushed up) After that, it's a piece of cake to support.

emonty committed:
Change to auth_type to match python-openstackclient

emonty committed:
Update common OpenStack requests-related parameters

Also, update docs related to earlier changes in this stack.

bcoca commented:
my worry here is that this will break existing modules, even if deprecated they should still work

emonty committed:
Add api timeout now that shade spports it everywhere

emonty commented:
@bcoca the existing deprecated modules do not use this function, they use the other one.

bcoca commented on lib/ansible/utils/module_docs_fragments/openstack.py:
```diff
@@ -69,12 +60,35 @@ class ModuleDocFragment(object):
+      - How long should the socket layer wait before timing out for API calls.
+        If this is omitted, nothing will be passed to the requests library.
+    required: false
+  verify:
```
is this the same as "validate_certs" in other modules? if so I would prefer to standardize the naming to that

bcoca commented on lib/ansible/utils/module_docs_fragments/openstack.py:
```diff
@@ -69,12 +60,35 @@ class ModuleDocFragment(object):
+    description:
+      - How long should the socket layer wait before timing out for API calls.
+        If this is omitted, nothing will be passed to the requests library.
+    required: false
```
non required fields should show default, even if just null

bcoca commented:
aside from the 2 comments everything else looks good

emonty committed:
Align verify parameter with validate_certs

The rest of ansible uses validate_certs, so make that the main
documented parameter. However, leave verify as an alias since that's the
passthrough value to the underlying libraries.
